### ISSUE TITLE
Inject Generated Sprites into Asset Pipeline

### ISSUE THREAD

**Issue by yfeldblum** [Compass/compass#643]
Right now, sprites are generated and stored in a file (using `@include "blah/*.png"` in a stylesheet, the sprite gets stored in a file like `public/assets/blah-shortdigest.png`).

When the application is using the asset pipeline, sprites should be added to the asset pipeline's cache. They should not be copied into the `public/assets` directory, and they should be targets for the asset pipeline to create new files with long digests (just as it does for all other images), and compass should use those long digests when `config.assets.digests == true`.

In other words, generated sprites should be injected into the asset pipeline and, once injected, be dealt with like any other image asset.

This would, e.g., permit configuring a webserver to add an expires-never header for all URLs like `^/assets/.*-[0-9a-z]{32}\.(js|css|png)$`. It would also prevent the `public/assets` directory from getting "polluted" in development/test with generated asset files.

**Comment by scottdavis** [comment]
I wish there was a better solution for this. we tried to get this working the correct way and for now the current implementation will have to stand until changes in sprockets allow this sort of behavior. If you want to try and tackle this again be my guest its been a up hill battle for sprockets integration.

**Comment by yfeldblum** [comment]
Ah. Is Sprockets missing an API for understanding that something in the pipeline wants to add a generated asset, and that the asset currently being compiled declares an additional dependency on that generated asset?

**Comment by scottdavis** [comment]
yes

**Comment by levity** [comment]
Is this still known to be broken? If so, maybe it should be called out more prominently somewhere in the documentation, as it's pretty painful to set up sprites only to find out in production that they don't work with digests.

**Comment by scottdavis** [comment]
They do work in production im using them with rails 3.2 in production just fine? what problem are you seeing?
